File: util/jester/jester/config/default.py
###########################################################################
## Copyright (C) Flowbox, Inc # All Rights Reserved
## Unauthorized copying of this file, via any medium is strictly prohibited
## Proprietary and confidential
## Flowbox Team <contact@flowbox.io>, 2013
###########################################################################

import logging
logger = logging.getLogger(__name__)

def init(ctx):
    ctx.toolchain = 'cabal'
    ctx.target    = 'native'

def configure(ctx):
    pass

def build(ctx):
    pass

def clean(ctx):
    pass

def postbuild(ctx):
    import os
    import shutil
    if ctx.target.name == 'js':
        main_path, _ = os.path.splitext(ctx.main)
        main = main_path + '.js'
        if not os.path.exists(main):
            logger.error("Result js file '%s' does not exist." % main)
        binname   = ctx.executable + '.js'
        buildpath = os.path.join('dist', 'build', binname)
        maindst = os.path.join(buildpath,binname)
        try:
            os.makedirs(buildpath)
        except:
            pass
        logger.debug("Moving result file to '%s'" % maindst)
        shutil.move(main, maindst)
    pass

File: src/Luna/ASTNew/Traversals/Default.hs
---------------------------------------------------------------------------
-- Copyright (C) Flowbox, Inc - All Rights Reserved
-- Unauthorized copying of this file, via any medium is strictly prohibited
-- Proprietary and confidential
-- Flowbox Team <contact@flowbox.io>, 2014
---------------------------------------------------------------------------
 {-# LANGUAGE UndecidableInstances #-}
 {-# LANGUAGE OverlappingInstances #-}

module Luna.ASTNew.Traversals.Default where

import           Flowbox.Prelude        hiding (Cons, Traversal, traverse)
import           Luna.ASTNew.Traversals.Class
import           Luna.ASTNew.Label      (Label(Label))


----- Basic -----
instance DefaultTraversal base m a => Traversal base m a where traverse = defaultTraverse
--instance DefaultTraversal base m a where defaultTraverse _ = pure


----- Label -----
instance Traversal base m a => Traversal base m (Label l a) where
    traverse base (Label l a) = fmap (Label l) $ traverse base a
